pkg/providers/instance/instance.go

```diff
@@ -166,6 +167,11 @@ func (p *Provider) Delete(ctx context.Context, id string) error {
 		}
 		return fmt.Errorf("terminating instance, %w", err)
 	}
+	for _, instance := range response.TerminatingInstances {
+		if *instance.CurrentState.Name != ec2.InstanceStateNameTerminated {
+			return cloudprovider.NewRetryableError(fmt.Errorf("terminating instance '%s'", aws.StringValue(instance.CurrentState.Name)))
```

Comment:
IMO this error should be more verbose about why it is "retryable". Maybe something like, `fmt.Errorf("received terminate instance call but instance is still in state %q, instance.CurrentState.Name")`.

Also, you can use the `%q` formatting directive to easily quote variables